api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -218,6 +233,15 @@ type NodePoolSpec struct {
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Arch is immutable"
 	// +optional
 	Arch string `json:"arch,omitempty"`
+
+	// imageType specifies the type of image to use for node instances.
+	// Valid values are ImageTypeDefault (empty string/default) or ImageTypeWindows ("windows").
+	// When set to ImageTypeWindows, the controller will automatically populate the AMI field
+	// with a Windows-compatible AMI based on the region and OpenShift version.
+	//
+	// +optional
+	// +kubebuilder:validation:Enum="";windows
+	ImageType ImageType `json:"imageType,omitempty"`
```

Comment:
This is meant to be an AWS platform specific thing? If so, it should probably go under AWSNodePoolPlatform.